File: LeetCode/Easy/2099-Find_Subsequence_Of_Length_K_With_The_Largest_Sum/2099-FindSubsequenceOfLengthKWithTheLargestSum.cpp
class Solution {
public:
    vector<int> maxSubsequence(vector<int>& nums, int k) {
        std::vector<std::pair<int, int>> indexedNums;

        for (int i = 0; i < nums.size(); ++i) {
            indexedNums.push_back({nums[i], i});
        }

         std::sort(indexedNums.begin(), indexedNums.end(), [](const std::pair<int, int>& a, const std::pair<int, int>& b) {
            return a.first > b.first; // Sort in descending order of value
        });

        std::vector<std::pair<int, int>> greatest;
        for (int i = 0; i < k; ++i) {
            greatest.push_back(indexedNums[i]);
        }

        std::sort(greatest.begin(), greatest.end(), [](const std::pair<int, int>& a, const std::pair<int, int>& b) {
            return a.second < b.second; // Sort in ascending order of original index
        });

        std::vector<int> ans;
        for (const auto& p : greatest) {
            ans.push_back(p.first);
        }

        return ans;
    }
};

File: LeetCode/Easy/2099-Find_Subsequence_Of_Length_K_With_The_Largest_Sum/2099-FindSubsequenceOfLengthKWithTheLargestSum.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var maxSubsequence = function (nums, k) {
    const indexedNums = nums.map((num, i) => ({ num, originalIndex: i }));

    const greatest = indexedNums
        .sort((a, b) => b.num - a.num) // Sort descending by number
        .slice(0, k); // Take the top k elements

    const idxGreatest = greatest.sort((a, b) => a.originalIndex - b.originalIndex);

    const ans = idxGreatest.map(item => item.num);

    return ans;
};